workflows_core/engine/cluster_engine.py

```diff
@@ -0,0 +1,50 @@
+import concurrent.futures
+
+import logging
+import traceback
+from typing import Any
+
+from workflows_core.engine.abstract_engine import AbstractEngine
+
+from tqdm.auto import tqdm
+
+logger = logging.getLogger(__file__)
+
+
+class ClusterEngine(AbstractEngine):
+    def __init__(self, show_progress_bar: bool = True, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self._show_progress_bar = show_progress_bar
+        self._progress = tqdm(desc=repr(self.operator), total=self.num_chunks * 2)
+
+    def apply(self) -> Any:
+
+        iterator = self.iterate()
+
+        documents = []
+        for chunk in iterator:
+            documents += chunk
+
+        new_batch = self.operator(documents)
+
+        payloads = []
+        for i in range(self._num_chunks):
```

Comment:
This would be faster if you used a generator instead of doing it in-memory but can be fixed later.

Reply:
added it anyways